lib/singlepass-backend/src/codegen_x64.rs

```diff
@@ -4769,6 +5093,82 @@ impl FunctionCodeGenerator<CodegenError> for X64FunctionCode {
                     self.machine.release_temp_gpr(tmp_out);
                 }
             }
+            Operator::I64TruncUSatF32 => {
+                let loc =
+                    get_location_released(a, &mut self.machine, self.value_stack.pop().unwrap());
+                let ret = self.machine.acquire_locations(
+                    a,
+                    &[(WpType::I64, MachineValue::WasmStack(self.value_stack.len()))],
+                    false,
+                )[0];
+                self.value_stack.push(ret);
+
+                let tmp_out = self.machine.acquire_temp_gpr().unwrap();
+                let tmp_in = self.machine.acquire_temp_xmm().unwrap(); // xmm2
+
+                Self::emit_relaxed_binop(
+                    a,
+                    &mut self.machine,
+                    Assembler::emit_mov,
+                    Size::S32,
+                    loc,
+                    Location::XMM(tmp_in),
+                );
+                Self::emit_f32_int_conv_check_sat(
+                    a,
+                    &mut self.machine,
+                    tmp_in,
+                    -1.0,
+                    18446744073709551616.0,
```

Comment:
here too, if these numbers are shared between the backends it might even make sense to put them in a shared location

Reply:
These aren't quite shared between backends, though they absolutely could be. Note that these are f32/f64 while the llvm-backend uses u64, also llvm-backend uses values that are inside the valid range while singlepass uses values outside the valid range. I don't think there's any underlying need for these differences, but it might be that these are expressed in the way most convenient for lowering code in each backend. I'm going to punt that investigation and possible cleanup to a future PR.